src/chainstate/burn/db/sortdb.rs

```diff
@@ -1131,6 +1131,12 @@ impl SortitionDB {
         db_tx.commit()?;
         Ok(())
     }
+
+    /// Load up all snapshots, in ascending order by block height.  Great for testing!
+    pub fn get_all_snapshots(&self) -> Result<Vec<BlockSnapshot>, db_error> {
```

Comment:
This function will not work if there's a PoX fork or a Bitcoin fork. Since this is something like a diagnostic function, maybe we should guard this with a compile flag (like a `diagnostics` feature?), so that we don't try to use this function elsewhere.

Reply:
Why won't it work?  If there's a Bitcoin fork, the orphaned snapshots should still be there.  In a PoX fork, we will probably be marking orphaned snapshots as invalid, so the caller can filter them out.

This method is only meant to be used by the chainstate-replay diagnostic tool at this time, and it can be extended to filter out invalidated PoX snapshots.

Reply:
Ah, okay, that makes sense -- as long as the caller can deal with orphaned snapshots, this works.